widgetssdk/src/test/java/com/glia/widgets/callvisualizer/ActivityWatcherControllerTest.kt

```diff
@@ -0,0 +1,243 @@
+package com.glia.widgets.callvisualizer
+
+import android.app.Activity
+import android.content.Intent
+import android.view.View
+import android.view.Window
+import androidx.activity.result.ActivityResultLauncher
+import com.glia.androidsdk.GliaException
+import com.glia.androidsdk.comms.MediaUpgradeOffer
+import com.glia.widgets.callvisualizer.controller.CallVisualizerController
+import com.glia.widgets.callvisualizer.domain.IsCallOrChatScreenActiveUseCase
+import com.glia.widgets.core.dialog.Dialog.*
+import com.glia.widgets.core.dialog.model.DialogState
+import com.glia.widgets.core.screensharing.ScreenSharingController
+import com.glia.widgets.view.head.controller.ServiceChatHeadController
+import junit.framework.TestCase.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.*
+import java.lang.ref.WeakReference
+
+internal class ActivityWatcherControllerTest {
+
+    private val callVisualizerRepository = mock(CallVisualizerRepository::class.java)
+    private val callVisualizerController = mock(CallVisualizerController::class.java)
+    private val serviceChatHeadController = mock(ServiceChatHeadController::class.java)
+    private val screenSharingController = mock(ScreenSharingController::class.java)
+    private val isCallOrChatActiveUseCase = mock(IsCallOrChatScreenActiveUseCase::class.java)
+    private val watcher = mock(ActivityWatcherContract.Watcher::class.java)
+    private val controller = ActivityWatcherController(callVisualizerController, screenSharingController, serviceChatHeadController)
+    private val activity = mock(Activity::class.java)
+    private val activityReference: WeakReference<Activity?> = WeakReference(activity)
+
+    @Before
+    fun setup() {
+        val window = mock(Window::class.java)
+        val view = mock(View::class.java)
+        whenever(activity.window).thenReturn(window)
+        whenever(window.decorView).thenReturn(view)
+        controller.setWatcher(watcher)
+        cleanup()
+        reset(watcher, callVisualizerController, callVisualizerRepository, serviceChatHeadController, screenSharingController)
+    }
+
+    @Test
+    fun `onActivityResumed call or chat is active`() {
```

Comment:
We had an [agreement](https://salemove.slack.com/archives/C01F9K88SQZ/p1677586771381839?thread_ts=1677153711.779559&cid=C01F9K88SQZ) recently that test name should have the following structure:

- First is the method that is being tested (e.g: "onActivityResumed")
- Second is the expected behavior of the test case (e.g: "completes with 0")
- Last is when that test case is applicable (e.g: "when error is returned")

It seems to me that we are missing the second part. That's why it's difficult to understand the test, at least for me.

Could you please make sure all tests in this PR have agreed name structure?

Reply:
Thank you, I missed the final conclusion of that thread